nav2_recoveries/include/nav2_recoveries/recovery.hpp

```diff
@@ -78,21 +73,22 @@ class Recovery
   // It's up to the derived class to define the final commanded velocity.
   virtual Status onCycleUpdate() = 0;
 
-protected:
-  rclcpp::Node::SharedPtr node_;
-  std::string recovery_name_;
-  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr vel_pub_;
-  tf2_ros::Buffer & tf_;
-  std::unique_ptr<ActionServer> action_server_;
+  // an opportunity for derived classes to do something on configuration
+  // if they chose
+  virtual void onConfigure()
+  {
+  }
 
-  std::shared_ptr<nav2_costmap_2d::CostmapSubscriber> costmap_sub_;
-  std::shared_ptr<nav2_costmap_2d::FootprintSubscriber> footprint_sub_;
-  std::unique_ptr<nav2_costmap_2d::CollisionChecker> collision_checker_;
-  double cycle_frequency_;
 
-  void configure()
+  void configure(
+    const rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+    const std::string & name, std::shared_ptr<tf2_ros::Buffer> tf) override
   {
-    RCLCPP_INFO(node_->get_logger(), "Configuring %s", recovery_name_.c_str());
+    RCLCPP_INFO(parent->get_logger(), "Configuring %s", name.c_str());
+
+    node_ = parent;
+    tf_ = tf;
```

Comment:
Cleanup the `node_` and `tf_` in the `cleanup()` call?

Reply:
node and tf are from the recovery server, which within that reset on cleanup. It wouldnt make sense for each of `N` recoveries to  reset the TF buffer or node ptr on shutdown

Reply:
I don't see where you are resetting the node and tf in the `on_cleanup` for the recovery server.